### PR TITLE
Feat: proto common

### PR DESCRIPTION
**Pull request by gluax**
## Motivation

To have the protobuf in a shared repo.

## Explanation of Changes

Creates a workspace.
Moves the `seda-common` to a workspace member.
Creates a new `seda-proto-common` as a workspace member.

## Testing

Works in the test suite using the same versions as `devnet`, which uses both.

Need to update the overlay yet.

## Related PRs and Issues

N/A

### CODE REVIEW

File: crates/proto-common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "seda-proto-common"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+# prost.workspace = true
+# prost-types.workspace = true
+prost = { version = "0.13", default-features = false }
+prost-types = { version = "0.13", default-features = false, optional = true }
```
Comment (Thomasvdam) [inline crates/proto-common/Cargo.toml]: I guess one of these two can go?

File: crates/proto-common/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "seda-proto-common"
+version = "0.1.0"
```
Comment (Thomasvdam) [inline crates/proto-common/Cargo.toml]: What we did for the TS variant is to have the package version exactly the same as the chain version from which the proto files were taken. I think that helps making things a little clearer when protobuf breaks down between components.

Comment (gluax) [reply crates/proto-common/Cargo.toml]: What if it's not tagged a new version yet?

Comment (Thomasvdam) [reply crates/proto-common/Cargo.toml]: Maybe add an additional prerelease/alpha suffix?

Comment (gluax) [reply crates/proto-common/Cargo.toml]: I coordinated with HY on what the next version should be and changed it to that :).

**Review by Thomasvdam — APPROVED** [review]

**Review by mariocao — APPROVED** [review]

**Pull request merged**